### PR TITLE
Fix issue 7551 - Regex parsing bug for right bracket in character class

### PR DESCRIPTION
**Pull request by DmitryOlshansky**

### CODE REVIEW

File: std/regex/internal/tests.d [hunk 1]
```diff
@@ -968,3 +968,12 @@ unittest
...
 
+// bugzilla 7551
+unittest
+{
+    auto r = regex("[]abc]*");
+    assert("]ab".matchFirst(r).hit == "]ab");
+    assertThrown(regex("[]"));
+    auto r2 = regex("[]abc--ab]*");
+    assert("]ac".matchFirst(r2).hit == "]");
+}
```
Comment (9il) [inline std/regex/internal/tests.d]: newline please

**Comment by 9il** [comment]
LGTM

File: std/regex/internal/tests.d (hunk 1 above)
Comment (DmitryOlshansky) [inline std/regex/internal/tests.d]: Done

**Comment by Hackerpilot** [comment]
Auto-merge toggled on

**Pull request merged**